spec/lang/step.md

```diff
@@ -505,15 +514,21 @@ impl<M: Memory> Machine<M> {
             locals.insert(local, p);
         }
 
+
+        // Set which block the function will return to (if any)
+        self.mutate_cur_frame(|frame| {
+            frame.next_block = next_block;
+            frame.next_stmt = Int::ZERO;
+        });
```

Comment:
I'm not sure I conceptually agree with this -- I quite deliberately had the caller's stack frame point to the `Call` operation while the call is active. This is needed, for example, to get reasonable backtraces: we want to know where in the caller we were when the call was made, not where we will be when the call returns.

Reply:
Indeed, for backtraces the call origin would need to be stored. A reasonable place might be the callee's CallerReturnInfo (probably renamed to just CallerInfo or something) . My understanding is still that this information is only needed for diagnostic reasons. It cannot affect the execution, right?